### DISCUSSION TITLE
Private ghcr image for runner

### DISCUSSION THREAD

**Discussion by usernamessh** [actions/actions-runner-controller#2251] [Questions] [answered]
Hello community,

we have the following setup:
- Source-code:                      www.github.our-enterprise.com/org-a/repo
- Runner image (private):     ghcr.io/org-b/my-private-image-win2019:latest

I need to use a private hosted ghcr image for the runner. With public hosted images it is working fine. But for the private image, of course, I need to provide username & token. The token here is a different token to "controller-manager.github_token".
I wonder, where and how else to provide this information? I have seen a property called "imagePullSecrets: []" in values.yaml, but I don't know if this is the right one and how the syntax should look like?

Actually I would expect something like this to insert in the below yaml, but it does not seem to work:

<pre>
  image: ghcr.io/owner/image
  credentials:
     username: ${{ github.actor }}
     password: ${{ secrets.github_token_chcr }}
</pre>

runnerdeployment.yaml:
<pre>
apiVersion: actions.summerwind.dev/v1alpha1
kind: RunnerDeployment
metadata:
  name: actionsrunner-win2019
spec:
  replicas: 10
  template:
    spec:
      image: ghcr.io/org-b/my-private-image-win2019:latest
      dockerdWithinRunnerContainer: true
      nodeSelector:
        kubernetes.io/os: windows      
      repository: arg-b/my-other-private-repo
      labels: 
        - ARC
        - windows
        - X64
  </pre>

**Comment by MarkusWillinger** [comment, marked as answer]
Hello,
here is the solution:
https://learn.microsoft.com/en-ca/samples/azure-samples/github-runner-on-aks/self-hosted-github-actions-runner-on-aks-azure-kubernetes-service-with-auto-scale-option/
